assets/js/services/index.tsx

```diff
@@ -1,25 +1,42 @@
+import * as JwtApi from "./jwtApi";
+
 const axios = require("axios").default;
 
-axios.interceptors.request.use(
+const redirect = (suffix: string) => {
+  window.location.href = `${window.location.origin}${suffix}`;
+};
+
+const axiosWithInterceptor = axios.create();
+
+axiosWithInterceptor.interceptors.request.use(
   (config) => {
-    const bearer = `Bearer ${localStorage.getItem("jwt")}`;
-    config.headers.Authorization = bearer;
-    return config;
+    return JwtApi.addJwtToHeader(config);
   },
   (error) => {
     Promise.reject(error);
   }
 );
 
-axios.interceptors.response.use(undefined, (error) => {
-  if ([401, 403].includes(error.response.status)) {
-    localStorage.removeItem("jwt");
+axiosWithInterceptor.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+    const status = error.response ? error.response.status : null;
+    if (status === 401) {
+      const refreshToken = JwtApi.getRefreshToken();
+      try {
+        const response = await axios.post("/auth/refresh", { refreshToken });
+        JwtApi.setJwt(response.data);
+        const updatedConfig = JwtApi.addJwtToHeader(error.response.config);
+        return axiosWithInterceptor(updatedConfig);
+      } catch (err) {
+        JwtApi.destroyTokens();
+        alert("Your refresh token has expired. Please log in again.");
```

Comment:
According to one of our previous talks, did you check alerts in `chakra-ui`?

Reply:
I'll discuss it with Magda tommorow

Reply:
This is a little larger than expected. The task was moved [here](https://membraneframework.atlassian.net/browse/LIVE-95)